File: examples/stringify.ts
#!/usr/bin/env node

import {
  create,
  FilterSet,
  interact,
  utils
} from '../src/index.js'

const fs: FilterSet = {
  Pandoc: function (doc) {
    const str = create.Str(utils.stringify(doc))
    doc.blocks = [create.Para([str])]
    return doc
  }
}

interact([fs])

File: examples/stringify.lua
local pandoc = require "pandoc"
pandoc.utils = require "pandoc.utils"

local function Pandoc(doc)
  local str = pandoc.Str(pandoc.utils.stringify(doc))
  doc.blocks = {pandoc.Para {str}}
  return doc
end

return {
  { Pandoc = Pandoc }
}
